miscellaneous/roboconf-dm-rest-client/src/test/java/net/roboconf/dm/rest/client/delegates/ApplicationWsDelegateTest.java

```diff
@@ -409,6 +410,61 @@ public void testAddInstance_inexstingParentInstance() throws Exception {
 	}
 
 
+	@Test
+	public void testRemoveInstance_success() {
+		// Delete the Tomcat instance.
+		final Instance tomcat = this.app.getTomcat();
+		final String tomcatPath = InstanceHelpers.computeInstancePath( this.app.getTomcat() );
+		this.client.getApplicationDelegate().removeInstance( this.app.getName(), tomcatPath );
+
+		// Check it is gone.
+		List<Instance> children = this.client.getApplicationDelegate().listChildrenInstances( this.app.getName(), InstanceHelpers.computeInstancePath( this.app.getTomcatVm() ), true );
+		Assert.assertFalse(children.contains( tomcat ));
+	}
+
+
+	@Test
+	public void testRemoveInstance_nonExistingInstance() {
+		try {
+			this.client.getApplicationDelegate().removeInstance( this.app.getName(), "/I-do-not-exist" );
+			Assert.fail( "Expecting exception" );
+		} catch ( UniformInterfaceException e ) {
+			// Not found!
+			Assert.assertEquals( 404, e.getResponse().getStatus() );
+		}
+	}
+
+
+	@Test
+	public void testRemoveInstance_nonExistingApplication() {
+		try {
+			this.client.getApplicationDelegate().removeInstance( "I-am-not-an-app", InstanceHelpers.computeInstancePath( this.app.getTomcat() ) );
+			Assert.fail( "Expecting exception" );
+		} catch ( UniformInterfaceException e ) {
+			// Not found!
+			Assert.assertEquals( 404, e.getResponse().getStatus() );
+		}
+	}
+
+
+	@Test
+	public void testResynchronize_success() {
+		this.client.getApplicationDelegate().resynchronize( this.app.getName() );
```

Comment:
Maybe this would be better ?
```java
try {
  this.client.getApplicationDelegate().resynchronize(…);
} catch (Exception e) {
  Assert.fail("Unexpected exception: " + e);
}
```